exporter/azureblobexporter/exporter.go

```diff
@@ -0,0 +1,141 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package azureblobexporter // import "github.com/observiq/bindplane-agent/exporter/azureblobexporter"
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"strings"
+	"time"
+
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/exporter"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+)
+
+// azureBlobExporter exports OTLP data as Azure blobs
+type azureBlobExporter struct {
+	cfg        *Config
+	blobClient blobClient
+	logger     *zap.Logger
+	marshaler  marshaler
+}
+
+// newExporter creates a new Azure Blob exporter
+func newExporter(cfg *Config, params exporter.CreateSettings) (*azureBlobExporter, error) {
+	blobClient, err := newAzureBlobClient(cfg.ConnectionString)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create blob client: %w", err)
+	}
+
+	return &azureBlobExporter{
+		cfg:        cfg,
+		blobClient: blobClient,
+		logger:     params.Logger,
+		marshaler:  newMarshaler(cfg.Compression),
+	}, nil
+}
+
+// Capabilities lists the exporter's capabilities
+func (a *azureBlobExporter) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{MutatesData: false}
+}
+
+// metricsDataPusher pushes metrics data to Azure Blob
+func (a *azureBlobExporter) metricsDataPusher(ctx context.Context, md pmetric.Metrics) error {
+	buf, err := a.marshaler.MarshalMetrics(md)
+	if err != nil {
+		return fmt.Errorf("failed to marshal metrics: %w", err)
+	}
+
+	blobName := a.getBlobName("metrics")
+
+	return a.uploadBuffer(ctx, blobName, buf)
+}
+
+// logsDataPusher pushes logs data to Azure Blob
+func (a *azureBlobExporter) logsDataPusher(ctx context.Context, ld plog.Logs) error {
+	buf, err := a.marshaler.MarshalLogs(ld)
+	if err != nil {
+		return fmt.Errorf("failed to marshal logs: %w", err)
+	}
+
+	blobName := a.getBlobName("logs")
+
+	return a.uploadBuffer(ctx, blobName, buf)
+}
+
+// traceDataPusher pushes trace data to Azure Blob
+func (a *azureBlobExporter) traceDataPusher(ctx context.Context, td ptrace.Traces) error {
+	buf, err := a.marshaler.MarshalTraces(td)
+	if err != nil {
+		return fmt.Errorf("failed to marshal traces: %w", err)
+	}
+
+	blobName := a.getBlobName("traces")
+
+	return a.uploadBuffer(ctx, blobName, buf)
+}
+
+// getBlobName formats the blob name based on the configuration and current time stamp
+func (a *azureBlobExporter) getBlobName(telemetryType string) string {
+	now := time.Now()
```

Comment:
I considered this but since this is meant to look like the S3 exporter I copied what they did [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/awss3exporter/s3_writer.go#L63) and just used `time.Now()`

Reply:
Huh. Feels weird that two separate machines with the same configuration could put data into buckets hours apart like that.

If we want the same behavior as the S3 exporter, then this I guess we're fine. I do question whether it's correct there too, though.

Reply:
I'm not apposed either way. I think UTC makes more sense. I'll fix it here.